stomp_plugins/src/noise_generators/cartesian_constraints_sampling.cpp

```diff
@@ -0,0 +1,249 @@
+  tool_goal_tolerance_.resize(CARTESIAN_DOF_SIZE);
+  double ptol = DEFAULT_IK_POS_TOLERANCE;
+  double rtol = DEFAULT_IK_ROT_TOLERANCE;
+  tool_goal_tolerance_ << ptol, ptol, ptol, rtol, rtol, rtol;
+
+
+  return configure(config);
+}
+
+bool CartesianConstraintsSampling::configure(const XmlRpc::XmlRpcValue& config)
+{
+  using namespace XmlRpc;
+
+  try
+  {
+    XmlRpcValue params = config;
+
+    // noise generation parameters
+    XmlRpcValue stddev_param = params["stddev"];
+
+    // check  stddev
+    if(stddev_param.size() != stddev_.size())
+    {
+      ROS_ERROR("%s the 'stddev' parameter has incorrect number of cartesian DOF (6)",getName().c_str());
+      return false;
+    }
+
+    // parsing parameters
+    for(auto i = 0u; i < stddev_param.size(); i++)
+    {
+      stddev_[i] = static_cast<double>(stddev_param[i]);
+    }
+  }
+  catch(XmlRpc::XmlRpcException& e)
+  {
+    ROS_ERROR("%s failed to load parameters",getName().c_str());
+    return false;
+  }
+
+  return true;
+}
+
+bool CartesianConstraintsSampling::setMotionPlanRequest(const planning_scene::PlanningSceneConstPtr& planning_scene,
+                 const moveit_msgs::MotionPlanRequest &req,
+                 const stomp_core::StompConfiguration &config,
+                 moveit_msgs::MoveItErrorCodes& error_code)
+{
+  bool succeed = setupNoiseGeneration(planning_scene,req,config,error_code) &&
+          setupRobotState(planning_scene,req,config,error_code);
+  return succeed;
+}
+
+bool CartesianConstraintsSampling::setupNoiseGeneration(const planning_scene::PlanningSceneConstPtr& planning_scene,
+                                               const moveit_msgs::MotionPlanRequest &req,
+                                               const stomp_core::StompConfiguration &config,
+                                               moveit_msgs::MoveItErrorCodes& error_code)
+{
+  using namespace Eigen;
+
+  // convenience lambda function to fill matrix
+  auto fill_diagonal = [](Eigen::MatrixXd& m,double coeff,int diag_index)
+  {
+    std::size_t size = m.rows() - std::abs(diag_index);
+    m.diagonal(diag_index) = VectorXd::Constant(size,coeff);
+  };
+
+  // creating finite difference acceleration matrix
+  std::size_t num_timesteps = config.num_timesteps;
+  Eigen::MatrixXd A = MatrixXd::Zero(num_timesteps,num_timesteps);
+  int num_elements = (int((ACC_MATRIX_DIAGONAL_INDICES.size() -1)/2.0) + 1)* num_timesteps ;
+  for(auto i = 0u; i < ACC_MATRIX_DIAGONAL_INDICES.size() ; i++)
+  {
+    fill_diagonal(A,ACC_MATRIX_DIAGONAL_VALUES[i],ACC_MATRIX_DIAGONAL_INDICES[i]);
+  }
+
+  // create and scale covariance matrix
+  Eigen::MatrixXd covariance = MatrixXd::Identity(num_timesteps,num_timesteps);
+  covariance = A.transpose() * A;
+  covariance = covariance.fullPivLu().inverse();
+  double max_val = covariance.array().abs().matrix().maxCoeff();
+  covariance /= max_val;
+
+  // preallocating noise data
+  raw_noise_ = Eigen::VectorXd::Zero(CARTESIAN_DOF_SIZE);
```

Comment:
Why are you prealocating here when you are not using it here. Do it in constructor of in `applyCartesianNoise` method